main.go

```diff
@@ -374,7 +374,7 @@ func main() {
 		fmt.Println(err)
 		os.Exit(1)
 	}
-	if err := rootCmd.Execute(); err != nil {
+	if err := fang.Execute(context.Background(), rootCmd); err != nil {
```

Comment:
Did you consider using this?

https://pkg.go.dev/os/signal#NotifyContext

Reply:
we could probably do that in fang I think 🤔 

something like `fang.WithNotifyContext(SIGINT, SIGTERM, etc...)` 🤔

Reply:
https://github.com/charmbracelet/fang/pull/48

Reply:
Is `os/signal.Notify` really useful here if we're not doing anything special?

The default Go behavior is to finish the app, and that's basically what we want here.

Added some comments on https://github.com/charmbracelet/fang/pull/48/files#r2172861345. Let me know if I'm missing anything.

Reply:
> The default Go behavior is to finish the app, and that's basically what we want here.

that's right, we should be using the context on the tea app as well - just pushed a fix for that 

ideally then, the app should gracefully quit as the context is canceled, and any other operations can be canceled as well (e.g. exec.Commands, http requests, etc)

Reply:
I think we are good here